src/loadimg/utils.py

```diff
@@ -41,30 +45,99 @@ def load_img(
         ansi_art = load_img("image.png", output_type="ansi")
         ```
     """
-    img, original_name = load(img, input_type)
-
-    if output_type == "pil":
-        return img
-    elif output_type == "numpy":
-        return np.array(img)
-    elif output_type == "str":
-        secure_temp_dir = tempfile.mkdtemp(prefix="loadimg_", suffix="_folder")
-        file_name = original_name or f"{uuid.uuid4()}.png"
-        path = os.path.join(secure_temp_dir, file_name)
-        img.save(path)
-        return path
-    elif output_type == "base64":
-        img_type = img.format or "PNG"
-        with BytesIO() as buffer:
-            img.save(buffer, format=img_type)
-            img_str = base64.b64encode(buffer.getvalue()).decode("utf-8")
-        return f"data:image/{img_type.lower()};base64,{img_str}"
-    elif output_type == "ascii":
-        return image_to_ascii(img)
-    elif output_type == "ansi":
-        return image_to_ansi(img)
+    try:
+        img, original_name = load(img, input_type)
+        
+        # Validate loaded image
+        is_valid, error_msg = validate_image(img)
+        if not is_valid:
+            raise ValueError(f"Invalid image: {error_msg}")
+
+        if output_type == "pil":
+            return img
+        elif output_type == "numpy":
+            return np.array(img)
+        elif output_type == "str":
+            secure_temp_dir = tempfile.mkdtemp(prefix="loadimg_", suffix="_folder")
+            file_name = original_name or f"{uuid.uuid4()}.png"
+            path = os.path.join(secure_temp_dir, file_name)
+            img.save(path)
+            return path
+        elif output_type == "base64":
+            img_type = img.format or "PNG"
+            with BytesIO() as buffer:
+                img.save(buffer, format=img_type)
+                img_str = base64.b64encode(buffer.getvalue()).decode("utf-8")
+            return f"data:image/{img_type.lower()};base64,{img_str}"
+        elif output_type == "ascii":
+            return image_to_ascii(img)
+        elif output_type == "ansi":
+            return image_to_ansi(img)
+        else:
+            raise ValueError(f"Unsupported output type: {output_type}")
+            
+    except Exception as e:
+        raise ValueError(f"Failed to load image: {str(e)}") from e
+
+
+def load_imgs(
+    imgs: Union[str, List[str], List[Union[str, bytes, np.ndarray, Image.Image]]],
+    output_type: Literal["pil", "numpy", "str", "base64", "ascii", "ansi"] = "pil",
+    input_type: Literal["auto", "base64", "file", "url", "numpy", "pil"] = "auto",
+    max_workers: int = 4,
```

Comment:
it really depends on the hardware here so let's just set this to a safe number for now
```suggestion
    max_workers: int = 1,
```

Reply:
Okey i got that. You are really right